pkg/planner/core/find_best_task.go

```diff
@@ -2472,6 +2473,25 @@ func convertToTableScan(ds *DataSource, prop *property.PhysicalProperty, candida
 		if prop.MPPPartitionTp != property.AnyType {
 			return base.InvalidTask, nil
 		}
+		if candidate.path.Index != nil && candidate.path.Index.VectorInfo != nil {
+			vectorInfo := candidate.path.Index.VectorInfo
+			distanceMetricPB, err := vectorDistanceMetricToPB(vectorInfo.DistanceMetric)
+			if err != nil {
+				return base.InvalidTask, err
+			}
+
+			ts.AnnIndexExtra = &VectorIndexExtra{
+				candidate.path.Index,
+				uint(ts.Table.Columns[candidate.path.Index.Columns[0].Offset].GetFlen()),
+				tipb.ANNQueryInfo{
+					QueryType:      tipb.ANNQueryType_OrderBy,
+					DistanceMetric: distanceMetricPB,
+					TopK:           math.MaxUint32,
+					ColumnName:     candidate.path.Index.Columns[0].Name.L,
+					ColumnId:       ts.Table.Columns[candidate.path.Index.Columns[0].Offset].ID,
```

Comment:
In the open-source repo, we use the IndexId instead of the ColumnId.

Reply:
Added